abracadabra/src/g4-mandatory/trajectory_map.cc

```diff
@@ -0,0 +1,17 @@
+#include "trajectory_map.hh"
+
+#include <G4VTrajectory.hh>
+
+std::map<int, G4VTrajectory*> trajectory_map::map_;
+
+
+G4VTrajectory* trajectory_map::Get(int trackId) {
+    std::map<int, G4VTrajectory*>::iterator it = map_.find(trackId);
+    if (it == map_.end()) return 0;
```

Comment:
+ `0` -> `nullptr`
+ It's depressing that, even in C++20, there seems to be no standard abstraction for this bloody obvious pattern. What a primitive language! Perhaps we should write our own ...

   ```c++
   template<class M, class K, class V>
   V find_or(const M& map, K const& key, const V& default_value) {
     auto found = map.find(key);
     if (found == end(map)) { return default_value; }
     else                   { return found->second; }
   }
   ```

   ... probably a decent start ... to make it fully robust, perfectly-forwarding, etc. would probably take a silly amount of effort.